model/job/periodic_parser.go

```diff
@@ -15,26 +15,36 @@ type PeriodicParser struct{}
 
 // PeriodicSpec is the result of a successful parsing
 type PeriodicSpec struct {
-	// TODO Frequency
-	// TODO DaysOfMonth
-	DaysOfWeek []int // a slice of acceptable days, from 0 for sunday to 6 for saturday
-	AfterHour  int   // an hour between 0 and 23
-	BeforeHour int   // an hour between 1 and 24
+	Frequency   FrequencyKind
+	DaysOfMonth []int // empty for *, or a slice of acceptable days (1 to 31)
+	DaysOfWeek  []int // a slice of acceptable days, from 0 for sunday to 6 for saturday
+	AfterHour   int   // an hour between 0 and 23
+	BeforeHour  int   // an hour between 1 and 24
 }
 
+// FrequencyKind is used to tell if a periodic trigger is weekly or monthly.
+type FrequencyKind int
+
+const (
+	MonthlyKind FrequencyKind = iota
+	WeeklyKind
+)
+
 // NewPeriodicParser creates a PeriodicParser.
 func NewPeriodicParser() PeriodicParser {
 	return PeriodicParser{}
 }
 
 // Parse will transform a string like "on monday" to a PeriodicSpec, or will
 // return an error if the format is not supported.
-func (p *PeriodicParser) Parse(periodic string) (*PeriodicSpec, error) {
+func (p *PeriodicParser) Parse(frequency FrequencyKind, periodic string) (*PeriodicSpec, error) {
 	fields := strings.Fields(periodic)
 	spec := PeriodicSpec{
-		DaysOfWeek: []int{0, 1, 2, 3, 4, 5, 6},
-		AfterHour:  0,
-		BeforeHour: 24,
+		Frequency:   frequency,
+		DaysOfMonth: []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28},
```

Comment:
There are no 29, 30 or 31 in February. If we created a crontab that run on the 30th, it won't run in February, which is something I would want to avoid for a monthly trigger.

Reply:
Ok, but what if I want to run a @monthly trigger on `28-31`? 
I think if we should mention in the doc that we don't support days after the 28